### PR TITLE
chore: measure delivery lag in gateway and processor

### PR DESCRIPTION
**Pull request by lvrach**
# Description

Introduce two metrics:
* "processor.pickup_delivery_lag_seconds"
* "gateway.pickup_delivery_lag_seconds"

that capture the lag since ingestion in diffrent stages of our pipeline

## Linear Ticket

resolves PIPE-1168

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
  - Introduced tracking of `pickup_delivery_lag_seconds` with specific tags to monitor delivery performance.

- **Improvements**
  - Enhanced configuration for tracking `pickup_delivery_lag_seconds` with more granular time intervals.

- **Tests**
  - Updated test cases to include new statistics tracking and validations.

### CODE REVIEW

File: processor/processor.go [hunk 1]
```diff
@@ -1615,6 +1617,11 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 		requestIP := gatewayBatchEvent.RequestIP
 		receivedAt := gatewayBatchEvent.ReceivedAt
 
+		proc.statsFactory.NewSampledTaggedStat("processor.pickup_delivery_lag_seconds", stats.TimerType, stats.Tags{
```
Comment (mihir20) [inline processor/processor.go:1620]:
```suggestion
		proc.statsFactory.NewSampledTaggedStat("processor.event_pickup_lag_seconds", stats.TimerType, stats.Tags{
```

**Review by mihir20 — APPROVED** [review]
LGTM

File: gateway/handle.go [hunk 2]
```diff
@@ -794,6 +794,11 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byt
 			continue
 		}
 
+		gw.stats.NewTaggedStat("gateway.pickup_delivery_lag_seconds", stats.TimerType, stats.Tags{
```
[the remark this answers is not in the document]
Comment (lvrach) [reply gateway/handle.go:797]: I mean to convey we measure the lag from gateway_pickup not the pickup_lag

[the remark this answers is not in the document]
Comment (mihir20) [reply gateway/handle.go:797]: This is metric records time from ingestion-svc till its received at gateway right?

File: runner/buckets.go [hunk 3]
```diff
@@ -100,5 +100,13 @@ var (
 		"router.kafka.batch_size": {
 			1, 5, 10, 25, 50, 100, 250, 500, 1000, 2500, 5000, 10000,
 		},
+		"processor.pickup_delivery_lag_seconds": {
+			// 0.1s, 0.5s, 1s, 5s, 1m, 5m, 10m, 30m, 1h, 12h, 24h
+			0.1, 0.5, 1, 5, 60, 300, 600, 1800, 3600, 12 * 3600, 24 * 3600,
+		},
+		"gateway.pickup_delivery_lag_seconds": {
```
Comment (fracasula) [inline runner/buckets.go:107]:
```suggestion
		"processor.event_pickup_lag_seconds": {
			// 0.1s, 0.5s, 1s, 5s, 1m, 5m, 10m, 30m, 1h, 12h, 24h
			0.1, 0.5, 1, 5, 60, 300, 600, 1800, 3600, 12 * 3600, 24 * 3600,
		},
		"gateway.event_pickup_lag_seconds": {
```

**Review by fracasula — APPROVED** [review]

**Pull request merged**